### PR TITLE
#9739 & #9760: Fix - Dependency management of widgets

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
This PR attempts to fix the issue with the dependency management of widgets

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix

## Issue

**What is the current behavior?**
- #9739 
- #9760 

**What is the new behavior?**
- The dependencies are correctly formulated and enforce the original requirement of dependency of the widget
- Fixed counter widget doesn't set fallback value when the value is null/undefined
- Fixed Layer/Map multi-selection doesn't work on Mac

### Test data
`/dashboard/47595`
`/dashboard/47592`
`/dashboard/47714`

### Screenshot
image

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Review by offtherailz — CHANGES_REQUESTED** [review]
The fix works, anyway the code have to be a little reorganized and cleaned up, or it is diffcoult to read and maintain. 
Please follow my suggestions @dsuren1 . 

I'm asking this for master only I think.

File: web/client/plugins/Dashboard.jsx [hunk 1]
```diff
@@ -107,20 +108,24 @@ const WidgetsView = compose(
...
-                        (editingWidget.widgetType !== "map" &&
-                            editingWidget.widgetType === "chart"
-                            ? (target.layer && editingWidget && editingWidget?.charts?.map(c => c?.layer?.name)?.includes(target.layer.name))
-                            : (target.layer && editingWidget.layer && target.layer.name === editingWidget.layer.name)
-                        || editingWidget.widgetType === "map") && !target.mapSync
-                ) && target.id !== editingWidget.id
+            (target) => {
+                if (target.widgetType === "map") {
+                    return target.id !== editingWidget.id;
+                }
```
[the remark this answers is not in the document]
Comment (dsuren1) [reply web/client/plugins/Dashboard.jsx:114]: Not required, as dependency selection is only allowed on `map` and `table` and both are checked here

**Comment by dsuren1** [comment]
@offtherailz 
In case of new chart widget (not **saved yet**), when connecting to a transitive depedent widget, the chart view of the current chart widget in wizard will not show updated view (i.e excludes filters from dependent widget) but upon save and back to edit it shows updated chart view and works fine from there on as the chart should have an `id` which `dependenciesToWidget` enhancer requires when generating `dependenciesMap`. Is it behavior okay?

As part of this PR, I have removed the requirement of the presence of an`id` in the enhancer to allow all widget types to get updated when connecting to a transitive dependent widget (i.e before saving the current widget)

**Example**
Try creating a widget (chart/counter) and connect to the map of this dashboard (`/dashboard/47592`)

**Comment by offtherailz** [comment]
> @offtherailz In case of new chart widget (not **saved yet**), when connecting to a transitive depedent widget, the chart view of the current chart widget in wizard will not show updated view (i.e excludes filters from dependent widget) but upon save and back to edit it shows updated chart view and works fine from there on as the chart should have an `id` which `dependenciesToWidget` enhancer requires when generating `dependenciesMap`. Is it behavior okay?
> 
> As part of this PR, I have removed just the requirement of the presence of an`id` alone in the enhancer to allow all widget types to get updated when connecting to a transitive dependent widget (i.e before the widget is saved for the first time)
> 
> **Example** Try creating a widget (chart/counter) and connect to the map of this dashboard (`/dashboard/47592`), observe the behavior with id and without id validation in `dependenciesToWidget` enhancer locally

@dsuren1 it looks correct the way you did, from the user point of view. The ID of the widget should not be required to be dependent, so also new widget while editing first time can have a consistent preview..

**Review by offtherailz — APPROVED** [review]

**Comment by tdipisa** [comment]
@ElenaGallo please test in DEV and let us know. there are 2 issues connected to this PR and one of them is also for 2023.02.02

**Comment by ElenaGallo** [comment]
Test passed, @dsuren1 please backport to 2023.02.xx. Thanks

**Comment by tdipisa** [comment]
@ElenaGallo probably nothing to be backported for this. I'm waiting confirmation from @dsuren1.

**Comment by dsuren1** [comment]
@tdipisa You are right. This fix doesn't need to be backported, a fix without traces has already been merged to [stable](https://github.com/geosolutions-it/MapStore2/pull/9804)

**Comment by tdipisa** [comment]
@dsuren1 thank you for the confirmation.

**Pull request merged**